### DISCUSSION TITLE
ipsec pool reservation taken by different user

### DISCUSSION THREAD

**Discussion by ptchuba** [strongswan/strongswan#1830] [Q&A] [answered]
In my setup I want a user to get the same IP always. I'm using attr-sql and the IPsec pool tool to add a file that has the mappings. My file looks like this:
```
172.25.0.70=vdi1@example.local
172.25.0.80=vdi2@example.local
172.25.0.81=asn1dn:#305b311c301a06035504030c137573657231406578616d706c652e6c6f636165410b3009060355040613025553310f300d06035504070c0641757374696e3110300e060355040a0c074578616d706c65310b3009060355040b0c024954
```
With the last line being for a Windows host. 
And I load the file using 

`ipsec pool --add sqlite-pool-v4 --addresses addr-list-v4.txt --timeout 24`

What I noticed is that if I try to connect a Windows host that has not been reserved an IP, it grabs an IP that is reserved for a different user and currently not used. e.g. if 172.25.0.70 is reserved but not leased out yet, the new user grabs the IP. 

What I want is for the new user to fail ip assignment due to the fact that they do not have a reserved IP and there are no unreserved IPs available. Is this possible?

Thanks.

**Comment by tobiasbrunner** [comment, marked as answer]
You can't configure a timeout if you want static leases.

**Reply by ptchuba** [reply]
Thanks Tobias, that fixed it!!
